lightning_action/models/video_segmenter.py

```diff
@@ -0,0 +1,944 @@
+
+    def _setup_metrics(self) -> None:
+        """Initialize torchmetrics for tracking performance.
+        
+        Creates separate metric instances for train and validation
+        to avoid metric state contamination.
+        """
+        num_classes = self.output_size
+
+        # Training metrics
+        self.train_accuracy = Accuracy(
+            task='multiclass', 
+            num_classes=num_classes, 
+            ignore_index=self.ignore_index,
+        )
+        self.train_f1 = F1Score(
+            task='multiclass', 
+            num_classes=num_classes, 
+            ignore_index=self.ignore_index,
+        )
+
+        # Validation metrics (separate instances)
+        self.val_accuracy = Accuracy(
+            task='multiclass', 
+            num_classes=num_classes, 
+            ignore_index=self.ignore_index,
+        )
+        self.val_f1 = F1Score(
+            task='multiclass', 
+            num_classes=num_classes, 
+            ignore_index=self.ignore_index,
+        )
+
+    @abstractmethod
+    def _build_model(self) -> None:
+        """Build the model architecture. Must be implemented by subclasses."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def forward(
+        self,
+        x: Float[torch.Tensor, 'batch sequence features'],
+    ) -> dict[str, torch.Tensor]:
+        """Forward pass. Must be implemented by subclasses."""
+        raise NotImplementedError
+
+    @typechecked
+    def compute_loss(
+        self,
+        outputs: dict[str, torch.Tensor],
+        targets: Int[torch.Tensor, 'batch sequence'],
+        stage: str = 'train',
+    ) -> tuple[torch.Tensor, dict[str, torch.Tensor]]:
+        """Compute loss and metrics for a batch.
+        
+        Handles TCN padding by trimming predictions and targets to the
+        "middle" region where we have full temporal context.
+        
+        Args:
+            outputs: Model outputs dict containing 'logits'.
+            targets: Ground truth labels, shape (B, T).
+            stage: Either 'train' or 'val' for metric selection.
+        
+        Returns:
+            Tuple of (loss, metrics_dict).
+        """
+        logits = outputs['logits']
+        
+        # Trim TCN padding regions (model can't make good predictions there)
+        if hasattr(self, 'tcn_padding') and self.tcn_padding > 0:
+            logits = logits[:, self.tcn_padding:-self.tcn_padding, :]
+            targets = targets[:, self.tcn_padding:-self.tcn_padding]
+        
+        # Flatten for cross-entropy
+        logits_flat = logits.reshape(-1, self.output_size)
+        targets_flat = targets.reshape(-1)
+
+        # Handle edge case where all targets are ignore_index
+        # This can happen with certain data splits
+        if torch.all(targets_flat == self.ignore_index):
+            # Return zero loss (but maintain gradient graph)
+            loss = logits_flat.sum() * 0.0
+            
+            with torch.no_grad():
```

Comment:
no need for this, if all targets_flat are ignore index then accuracy and F1 aren't meaningul; can set both to 1